File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        std::vector<int> ans;
        int comp=0;
        for(int i =0; i<n+1;i++){
            if(i==0){
                ans.push_back(i);
                continue;
            }
            if((i & (i - 1)) == 0){
                comp = i;
                ans.push_back(1);
            }
            else{
                ans.push_back(1+ans[i-comp]);
            }
        }
        
        return ans;
    }
};

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = []
        for i in range(0, n+1):
            x = 1
            count = 0
            while x <= i:
                # print(x,i,count)
                if x & i:
                    count+=1
                x = x << 1
            res.append(count)
            
        return res